scripts/customize_conda.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euxo pipefail
+
+# This customization-script can be used to customize the conda environment.
+# It expects the following metadata:
+#   conda-component: (Required) Must be either ANACONDA or MINICONDA3.
+#   conda-env-config-uri: (Optional) Must be a gsutil path to the yaml config
+#   file.
+#   conda-packages: (Optional) A list of conda packages with versions to be
+#   installed in the base environment. Must be of the format
+#   <pkg1>:<version1>_<pkg2>:<version2>...
```

Comment:
Can we replace `_` with `,` or `;`? `_` is not as common as the others for delimiter.

Reply:
`,` and `;` are both special characters. We can not use them in metadata, as specified in the documentation - https://github.com/GoogleCloudDataproc/custom-images/blob/master/README.md#optional-arguments

Reply:
In this case, prefer `|` over `_`, the former is more commonly used for separation.

Reply:
`|` is also considered as a special character.

Reply:
Why? Could you elaborate? The problem is that `_` could be part of the package name.

Reply:
I see. Metadata argument does not handle escaping special characters. They have written in the documentation that if the value contains a character which needs escaping, we should encode the value, and decode it in our script. We can use `#` as a separator, it's not a special character and is not used in package names as well.